### PR TITLE
session only cookies patch - rev 01

### PR DESCRIPTION
**Pull request by uazo**
as https://github.com/bromite/bromite/issues/142#issuecomment-669246681

### CODE REVIEW

**Comment by csagan5** [comment]
Thanks for your work! See my reply on #142

File: build/patches/Session-only-cookies-v2.patch [hunk 1]
```diff
@@ -0,0 +1,775 @@
...
+diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/FourStateCookieSettingsPreference.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/FourStateCookieSettingsPreference.java
+old mode 100644
+new mode 100755
+index f2353a6ba2..5eb5f7ef58
+--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/FourStateCookieSettingsPreference.java
++++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/FourStateCookieSettingsPreference.java
+@@ -31,6 +31,7 @@ public class FourStateCookieSettingsPreference
+         BLOCK_THIRD_PARTY_INCOGNITO,
+         BLOCK_THIRD_PARTY,
+         BLOCK
```
Comment (csagan5) [inline build/patches/Session-only-cookies-v2.patch:196]:
```suggestion
         BLOCK,
```

File: build/patches/Session-only-cookies-v2.patch [hunk 2]
```diff
@@ -0,0 +1,775 @@
...
++        ,SESSION_ONLY
```
Comment (csagan5) [inline build/patches/Session-only-cookies-v2.patch:197]:
```suggestion
+        SESSION_ONLY
```

File: build/patches/Session-only-cookies-v2.patch [hunk 3]
```diff
@@ -0,0 +1,775 @@
...
+     }
+ 
+     /**
+@@ -48,6 +49,9 @@ public class FourStateCookieSettingsPreference
+         public boolean cookiesContentSettingEnforced;
+         //  Whether third-party blocking is enforced.
+         public boolean thirdPartyBlockingEnforced;
++
```
Comment (csagan5) [inline build/patches/Session-only-cookies-v2.patch:205]:
```suggestion
```

**Comment by csagan5** [comment]
@uazo this is the build with this patch: https://fdroid.bromite.org/fdroid/bromite/brm_sess_only.apk

I have also updated your branch with the adjusted version I built.

I do not use any particular tool to manage the patches, just `git am`, `git cherry-pick`, `git rebase`; the best would be to learn and use [quilt](https://wiki.debian.org/UsingQuilt).

**Comment by csagan5** [comment]
@uazo does session-only also block third-party cookies? Do you think we could make it work like that? If yes I would also update the description and move it before `Block all cookies`

**Comment by uazo** [comment]
> does session-only also block third-party cookies? 

Yes, chromium code block third-party cookies if a cookies is session-only.
You can see here https://github.com/bromite/bromite/pull/671/files#diff-2fad0c6caf1f9396df1464bde53f7a30R278

>  If yes I would also update the description and move it before Block all cookies

please tell me what exactly to write, my english is pretty poor.
then I edit them in the code and run grit (id for the translations)

**Comment by csagan5** [comment]
@uazo I have updated the description in this patch and moved it accordingly, thanks for confirming.

**Pull request merged**